framework/decode/vulkan_virtual_swapchain.h

```diff
@@ -118,15 +106,32 @@ class VulkanVirtualSwapchain : public VulkanSwapchain
     {}
 
   private:
-    VkResult CreateSwapchainImage(const DeviceInfo*               device_info,
-                                  const VkImageCreateInfo&        image_create_info,
-                                  SwapchainKHRInfo::VirtualImage& image);
-
-    int32_t FindFirstPresentSrcLayout(const VkRenderPassCreateInfo* create_info) const;
-
-    int32_t FindFirstPresentSrcLayout(const VkRenderPassCreateInfo2* create_info) const;
-
-    int32_t FindFirstPresentSrcLayout(uint32_t count, const VkImageMemoryBarrier* barriers) const;
+    // VirtualSwapchain-specific private information.
+    struct VirtualImage
+    {
+        VkDeviceMemory                        memory{ VK_NULL_HANDLE };
+        VkImage                               image{ VK_NULL_HANDLE };
+        VulkanResourceAllocator::MemoryData   memory_allocator_data{ 0 };
+        VulkanResourceAllocator::ResourceData resource_allocator_data{ 0 };
+    };
+
+    struct PrivateData
+    {
+        std::vector<VkCommandPool>                copy_command_pools;
```

Comment:
Are these all indexed by queue index or queue family index?  Could they be part of a subobject?  Then you could even put them in a map and not worry about resizing all these vectors, etc.

Reply:
I modified it to have a map in the `SwapchainData` associated per queue.  This will be ordered and should only be created for Queue families of interest.

Reply:
Done